model/distro/distro.go

```diff
@@ -581,6 +581,24 @@ func (d *Distro) GetProviderSettingByRegion(region string) (*birch.Document, err
 	return nil, errors.Errorf("distro '%s' has no settings for region '%s'", d.Id, region)
 }
 
+// ReplaceProviderSetting replaces the region's settings document in-memory
+func (d *Distro) ReplaceProviderSetting(region string, newSettings *birch.Document) error {
+	index := -1
+	for i, s := range d.ProviderSettingsList {
+		if val, ok := s.Lookup("region").StringValueOK(); ok && val == region {
+			index = i
+			break
```

Comment:
I think this function isn't necessary given the earlier comment, but if you do choose to keep it you don't need to use index: you can put line 597 on line 589 and then return, and return the error outside the loop

Reply:
removed